framework/src/ble_manager/ble_manager_autoconnect.c

```diff
@@ -0,0 +1,249 @@
+			if (evt == BLE_AUTOCON_EVT_SCAN_START) {
+				state = BLE_AUTOCON_STATE_SCAN_STARTING;
+			}
+		} else if (state == BLE_AUTOCON_STATE_SCAN_STARTING) {
+			if (evt == BLE_AUTOCON_EVT_MAC_SCANNED) {
+				state = BLE_AUTOCON_STATE_SCAN_STOPPING;
+				ret = ble_client_stop_scan();
+				if (ret != BLE_MANAGER_SUCCESS) {
+					BLE_LOG_ERROR("[BLEMGR] auto conn scan stop fail[%d]\n", ret);
+					goto finish_auto;
+				}
+			}
+		} else if (state == BLE_AUTOCON_STATE_SCAN_STOPPING) {
+			if (evt == BLE_AUTOCON_EVT_SCAN_STOP) {
+				state = BLE_AUTOCON_STATE_CONNECTING;
+				ret = ble_client_reconnect(ctx);
+				if (ret != BLE_MANAGER_SUCCESS) {
+					BLE_LOG_ERROR("[BLEMGR] auto conn reconnect fail[%d]\n", ret);
+					goto finish_auto;
+				}
+			}
+		} else if (state == BLE_AUTOCON_STATE_CONNECTING) {
+			if (evt == BLE_AUTOCON_EVT_CONNECTED) {
+				state = BLE_AUTOCON_STATE_CONNECTED;
+				break;
+			} else if (evt == BLE_AUTOCON_EVT_DISCONNECTED) {
+				state = BLE_AUTOCON_STATE_DISCONNECT;
+			}
+		}
+		BLE_LOG_DEBUG("[BLEMGR] auto conn state : %d\n", state);
+	}
+finish_auto:
+	mq_close(ctx->recv_mqfd);
+	mq_close(ctx->send_mqfd);
+	ctx->recv_mqfd = 0;
+	ctx->send_mqfd = 0;
+	mq_unlink(mq_name);
+	g_ctx = NULL;
+
+	BLE_LOG_INFO("[BLEMGR] Auto Connection Done\n");
+
+	return NULL;
+}
+
+ble_autocon_result_e ble_manager_autoconnect(ble_client_ctx *ctx)
+{
+	if (ctx->state != BLE_CLIENT_AUTOCONNECTING) {
+		return BLE_AUTOCON_INVALID_STATE;
+	}
+
+	pthread_attr_t attr;
+	struct sched_param sparam;
+	int status;
+	pthread_t g_thread = 0;
+
+	/* Initialize the attribute variable */
+	status = pthread_attr_init(&attr);
+	if (status != 0) {
+		BLE_LOG_ERROR("[BLEMGR] pthread_attr_init failed, status=%d\n", status);
+		return BLE_AUTOCON_FAIL;
+	}
+
+	/* 1. set a priority */
+	sparam.sched_priority = 100;
+	status = pthread_attr_setschedparam(&attr, &sparam);
+	if (status != 0) {
+		BLE_LOG_ERROR("[BLEMGR] pthread_attr_setschedparam failed, status=%d\n", status);
+		return BLE_AUTOCON_FAIL;
+	}
+
+	/* 2. set a stacksize */
+	status = pthread_attr_setstacksize(&attr, 1024);
+	if (status != 0) {
+		BLE_LOG_ERROR("[BLEMGR] pthread_attr_setstacksize failed, status=%d\n", status);
+		return BLE_AUTOCON_FAIL;
+	}
+
+	/* 3. set a sched policy */
+	status = pthread_attr_setschedpolicy(&attr, SCHED_RR);
+	if (status != 0) {
+		BLE_LOG_ERROR("[BLEMGR] pthread_attr_setschedpolicy failed, status=%d\n", status);
+		return BLE_AUTOCON_FAIL;
+	}
+
+	/* 4. create pthread with entry function */
+	status = pthread_create(&g_thread, &attr, _autocon_process, (pthread_addr_t)ctx);
+	if (status != 0) {
+		BLE_LOG_ERROR("[BLEMGR] pthread_create failed, status=%d\n", status);
+		return BLE_AUTOCON_FAIL;
+	}
+
+	status = pthread_setname_np(g_thread, "ble_autoconnect");
+	if (status != 0) {
+		BLE_LOG_ERROR("[BLEMGR] failed to set name for pthread(%d)\n", status);
+		return BLE_AUTOCON_FAIL;
+	}
+
+	/* Wait for the threads to stop */
+	pthread_detach(g_thread);
```

Comment:
`pthread_detach` cannot wait the threads to stop.
If you want to wait, then you should use `pthread_join` instead of `pthread_detach`.
`pthread_detach` only deallocates the resource when that pthread terminates.

Reply:
@jeongchanKim 
That's my mistake. I will remove the comment